src/rdydata.c

```diff
@@ -198,71 +198,76 @@ PetscErrorCode RDySetWaterSourceForRegion(RDy rdy, const PetscInt region_idx, Pe
 
   if (region.num_cells) {
     PetscReal *source_values;
-    PetscCall(PetscCalloc1(rdy->mesh.num_owned_cells, &source_values));
+    PetscCall(PetscCalloc1(region.num_cells, &source_values));
```

Comment:
I think we should allocate memory for `mesh.num_owned_cells`. This is because the source-sink term won't be applied to the ghost cells. Also, we need to check if `region.num_cells` corresponds to the global cells in the region or local+ghost cell.

Btw, we can introduce a new member variable `region.num_owned_cells` and use it to allocate memory.

Reply:
I just realized that `mesh.num_owned_cells` isn't the correct choice. We should introduce a new member variable, `region.num_owned_cells`, and use it to allocate memory using it.